### PR TITLE
Removed Unnecessary Inline Declarations

### PR DESCRIPTION
**Pull request by ShikharJ**
Relevant: #1265

### CODE REVIEW

**Comment by isuruf** [comment]
Not all `inline` are unnecessary.

File: symengine/series.h [hunk 1]
```diff
@@ -349,14 +346,13 @@ class SeriesBase : public SeriesCoeffInterface
         }
     }
 
-    static inline Poly series_cot(const Poly &s, const Poly &var,
-                                  unsigned int prec)
+    static Poly series_cot(const Poly &s, const Poly &var, unsigned int prec)
```
Comment (isuruf) [inline symengine/series.h]: These functions are not member functions.

Comment (ShikharJ) [inline symengine/series.h]: But isn't this defined inside class `SeriesBase`? And as far as I can recall templates are also automatically inlined. Am I missing something?

Comment (rdbyk) [inline symengine/series.h]: @ShikharJ these are `static` member functions, they are afaik like regular functions.

Comment (ShikharJ) [inline symengine/series.h]: @rdbyk Please refer [here](http://stackoverflow.com/questions/20550459/is-a-static-member-function-defined-inside-a-class-implicitly-inlined).

Comment (rdbyk) [inline symengine/series.h]: @ShikharJ Sorry for the noise, you are right!

**Comment by ShikharJ** [comment]
Ping @isuruf.

**Comment by ShikharJ** [comment]
Ping @isuruf.

**Comment by ShikharJ** [comment]
Ping @isuruf.

**Comment by ShikharJ** [comment]
Ping @isuruf.

**Comment by isuruf** [comment]
I'll have to go through the code a little bit before merging.

**Comment by ShikharJ** [comment]
Just to be sure, regarding moving the implementations to the `cpp` files, we'd want to move only the definitions of the member functions right?

**Comment by ShikharJ** [comment]
@isuruf Anything on this?

**Comment by rikardn** [comment]
In C++20 inline is no longer implicit in modules. To be future compatible I think we should keep inline declarations. I would vote for closing this PR

**Pull request closed without merging**